contrib/google.golang.org/grpc/option.go

```diff
@@ -1,13 +1,45 @@
 package grpc
 
-type interceptorConfig struct{ serviceName string }
+// The StreamMode is used to control what is traced in the
+// StreamClientInterceptor and StreamServerInterceptor
+type StreamMode byte
```

Comment:
While I really love this pattern, I feel it's overkill here. I find it works well when there are many options and many combinations of them, while in this case we have only two:

* Trace calls
* Trace messages

The third one is implied when both are set. I'd dumb it down completely from an API standpoint, getting rid of bitwise operations and simply having options such as `grpctrace.WithStreamCalls` and `grpctrace.WithStreamMessages`. Much simpler IMO, what do you think?